handlers/hierarchy.go

```diff
@@ -135,13 +135,25 @@ func (f *Filter) HierarchyUpdate(w http.ResponseWriter, req *http.Request) {
 			continue
 		}
 
-		if err := f.FilterClient.AddDimensionValue(req.Context(), userAccessToken, "", collectionID, filterID, name, k); err != nil {
-			log.Event(ctx, "failed to add dimension value", log.Error(err))
-		}
+		mapUniqueOptions[k] = struct{}{}
 	}
 
-	wg.Wait()
-	http.Redirect(w, req, redirectURI, 302)
+	// Efficiently generate slice from map keys:
+	patchedOptionsValues := make([]string, len(mapUniqueOptions))
+	i := 0
```

Comment:
Is it necessary to use a counter? Can we not use append function to `patchedOptionsValues`?

The other thing might be to handle if the counter exceeds `len(mapUniqueOptions)`  to throw an error and return internal server error; then we would want to keep the count variable in `i`

Reply:
I did it this way to increase efficiency, but I think it's not necessary - I will use `append`